### DISCUSSION TITLE
How would one search the virtualized list?

### DISCUSSION THREAD

**Discussion by nemonemi** [TanStack/virtual#481] [General] [answered]
Let's say a user expects to hit CTRL+F and find an item that has not yet been rendered, how should this be handled with a virtualized list?

**Comment by agupta1989** [comment]
I don't think you can do that with virtualized list. However there can be different way to search the list such as adding a search text field.

**Comment by is-it-ayush** [comment, marked as answer]
Hi @nemonemi! You should call a callback to find the item index in your data array. Then you can use [scrollToIndex()](https://tanstack.com/virtual/v3/docs/api/virtualizer#scrolltoindex) from your `virtualizer` index as `virtualizer.scrollToIndex(fooundIndex)`. This is from top of my head & not tested but should work.

**Reply by dev-kml** [reply]
This is the approach I have taken and worked. 

The option `align: center` in `scrollToIndex` API doesn't seem to align the targeted node to the center of the viewport though.
It's a bit tricky based on the position of the targeted node as well. (i.e. if the node is closer to the bottom of the page, etc)

**Reply by is-it-ayush** [reply]
Glad! I haven't "tested" aligns in `scrollToIndex()` function but I believe you can also do some visual tricks such as highlighting the "targeted node" after it's been scrolled to make your intention more clear to end user.

**Reply by dev-kml** [reply]
Yup! That was a part of my implementation as well. (with css `keyframes` animations)
